src/main/java/org/spongepowered/api/command/args/parsing/QuotedStringTokenizer.java

```diff
@@ -67,19 +65,11 @@
 
         final TokenizerState state = new TokenizerState(arguments, lenient);
         List<SingleArg> returnedArgs = new ArrayList<>(arguments.length() / 4);
-        if (this.trimTrailingSpace) {
-            skipWhiteSpace(state);
-        }
         while (state.hasMore()) {
-            if (!this.trimTrailingSpace) {
-                skipWhiteSpace(state);
-            }
+            skipWhiteSpace(state);
             int startIdx = state.getIndex() + 1;
             String arg = nextArg(state);
-            returnedArgs.add(new SingleArg(arg, startIdx, state.getIndex()));
-            if (this.trimTrailingSpace) {
-                skipWhiteSpace(state);
-            }
+            returnedArgs.add(new SingleArg(arg, startIdx, arg.isEmpty() ? startIdx : state.getIndex()));
```

Comment:
This is where a comment would be handy to explain why you need to include this edge case.

That said, you say this is at the end of the string? Shouldn't `nextArg(state)` deal with this and return the correct state, rather than hacking it together here if `state.hasMore()` is saying that it has more to return?

Reply:
To be perfectly honest I have no idea; I saw the indexes were off and traced through this enough to verify that this was indeed a solution. I'd have to go back through and figure out exactly what's happening here. I believe implementing more consistent behavior would require a bit of a rework of the parser behind it.